inst/pages/exercises.qmd

```diff
@@ -1674,72 +1690,113 @@ p <- Heatmap(
 p
 ```
 
-**Extra** Cool Exercise : Identify Patterns and Clusters
-To make the exercise more engaging, let's do an exploratory data analysis task:
+**Extra** Cool exercises :
+
+1. Create an interactive heatmap. 
 
-#### 1. Cluster Analysis on PCoA/PCA Results:
+```{r}
+#| label: extra-interactiveheatmap-ex1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
 
-i. Perform clustering on the ordination results (e.g., k-means clustering).
-ii. Visualize the clusters on the ordination plots.
+library(mia)
+library(heatmaply)
 
-```{r, eval = FALSE}
-library(scater)
+data("GlobalPatterns", package = "mia")
 
-# Setting seed for reproducibility
-set.seed(123)
+# Create the interactive heatmap using aggregated data
+heatmaply(assay(agglomerateByRank(GlobalPatterns, rank = "Phylum")),
+          xlab = "Samples", ylab = "Phyla")
+```
+Notice we aggregate the "GlobalPatterns" dataset
+to a higher taxonomic level to make the data more manageable.
 
```

Comment:
This example should be structured. Now it is little bit hard to read. Moreover, usually it might be good idea to scale values as now we can only observe those values that are large (other values are just purple).

```
# Agglomerate
# Transform CLR
# Transform Z-score
# Plot heatmap
```